src/app/shared/markdown-wrapper/markdown-wrapper.service.ts

```diff
@@ -97,4 +98,17 @@ export class MarkdownWrapperService {
       FORBID_ATTR: this.forbidAttr,
     });
   }
+
+  /**
+   * Removes tab characters from markdown table headers or they won't display properly
+   * Markdown uses pipes with three or more hyphens in between to create columns and headers, and colons to align text
+   * E.g., |---|---|---| or | :--- | ---: | where there can be spaces between pipes and hypens but not tabs
+   * @param {string} data A string containing the markdown data
+   * @returns {string}
+   */
+  removeTabsFromTableHeaders(data: string): string {
+    return data
+      .replace(/(?<=\|[\t ]*)\t(?=[\t ]*:?-+:?[\t ]*\|)/g, '    ') // remove tabs after the starting pipe before the hyphens
+      .replace(/(?<=\|[\t ]*:?-+:?[\t ]*)\t(?=[\t ]*\|)/g, '    '); // remove tabs after the hyphens before the ending pipe
+  }
```

Comment:
I changed the regex a bit so it's more specific (and is potentially less likely to modify other content) but I wonder if it's become too complicated. The comments generally describe what's happening but on line 111 (and similarly for line 112), for example, tabs are removed when:
1. There is a preceding pipe `?<=\|` followed by zero or more spaces/tabs `[\t ]*`
2. And the tab comes before zero or more spaces/tabs `?=[\t ]*`, followed by zero or one colon`:?`, followed by more than one hyphen`-+`, followed by zero or one colon `:?`, followed by zero or more spaces/tabs `[\t ]*`, followed by a pipe `\|`

Also, to be safe the tab is being replaced by four spaces instead of an empty string.